src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/WarningPropertiesCollection.cs

```diff
@@ -153,12 +153,17 @@ public static bool ApplyProjectWideNoWarnProperties(ILogMessage message, Warning
         /// <param name="message">Message which should be upgraded to error if needed.</param>
         public static void ApplyProjectWideWarningsAsErrorProperties(ILogMessage message, WarningProperties warningProperties)
         {
+            if (message is null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
             if (message.Level == LogLevel.Warning && warningProperties != null)
             {
-                if ((warningProperties.AllWarningsAsErrors && message.Code > NuGetLogCode.Undefined) ||
+                if ((warningProperties.AllWarningsAsErrors && message.Code > NuGetLogCode.Undefined && !warningProperties.WarningsNotAsErrors.Contains(message.Code)) ||
```

Comment:
Should we check overlaps between `warningProperties.WarningsNotAsErrors` and `warningProperties.WarningsAsErrors` and warn customers about conflicting logic.
Let's say `NU1015` is both of above, then customer don't any error about it with current logic, still it's `warn as error` list too.

Reply:
According to spec it's upgraded to error, right?
![image](https://user-images.githubusercontent.com/8766776/196806861-bc26d481-9f4a-40d6-9ef2-927c176dc19d.png)

Reply:
> Should we check overlaps between `warningProperties.WarningsNotAsErrors` and `warningProperties.WarningsAsErrors` and warn customers about conflicting logic. Let's say `NU1015` is both of above, then customer don't any error about it with current logic, still it's in `warn as error` list too, then doing nothing anything is kind of awkward.

The compiler & msbuild don't really check that. I think that's an overkill at this point.

Reply:
> According to spec it should upgraded to error, right? ![image](https://user-images.githubusercontent.com/8766776/196806861-bc26d481-9f4a-40d6-9ef2-927c176dc19d.png)

Yes, I have a test case for that. 
https://github.com/NuGet/NuGet.Client/pull/4846/files#diff-31670e4347c7f45593fb021ea00abdd489a42ff00f70e17c364e655985c6953bR775

Note the `||` on the next line `warningProperties.WarningsAsErrors.Contains(message.Code)`

Reply:
Ok. Resolving.